front/src/applications/stdcmV2/components/StdcmConsist.tsx

```diff
@@ -0,0 +1,139 @@
+import { useStoreDataForRollingStockSelector } from 'modules/rollingStock/components/RollingStockSelector/useStoreDataForRollingStockSelector';
+import useFilterRollingStock from 'modules/rollingStock/hooks/useFilterRollingStock';
+import type { StdcmConfSliceActions } from 'reducers/osrdconf/stdcmConf';
+import { useAppDispatch } from 'store';
+
+import StdcmCard from './StdcmCard';
+import StdcmSuggestions from './StdcmSuggestions';
+
+interface StdcmSuggestionsConsistOption
+  extends SelectOptionObject,
+    Omit<LightRollingStockWithLiveries, 'id'> {
+  value: LightRollingStockWithLiveries;
+}
+
+const ConsistCardTitle = ({
+  rollingStock,
+}: {
+  rollingStock?: LightRollingStockWithLiveries | null;
+}) => {
+  if (!rollingStock) return null;
+
+  return (
+    <div className="stdcm-v2-consist-img w-75 d-flex justify-content-end">
+      <RollingStock2Img rollingStock={rollingStock} />
+    </div>
+  );
+};
+
+const StdcmConsist = ({ isPending = false }: { isPending?: boolean }) => {
+  const { speedLimitByTag, speedLimitsByTags, dispatchUpdateSpeedLimitByTag } =
+    useStoreDataForSpeedLimitByTagSelector();
+
+  const { updateRollingStockID } = useOsrdConfActions() as StdcmConfSliceActions;
+  const dispatch = useAppDispatch();
+
+  const { rollingStock } = useStoreDataForRollingStockSelector();
+
+  const [isSelectChanged, setIsSelectChanged] = useState(false);
+
+  const { filters, searchMateriel, filteredRollingStockList } = useFilterRollingStock();
+
+  const getLabel = (rs: LightRollingStockWithLiveries) => {
+    let res = '';
+    const { metadata, name } = rs;
+
+    const series = metadata?.series ?? (metadata?.reference || '');
+    const subseries =
+      metadata?.series && metadata.series !== metadata.subseries
+        ? metadata.subseries
+        : metadata?.detail || '';
+
+    if (series) res += series;
+    if (subseries) res += series ? ` (${subseries})` : subseries;
+    if (name) res += ` -- ${name}`;
+    return res;
+  };
+
+  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    searchMateriel(e.target.value);
+  };
+
+  const onInputOnBlur = () => {
+    // avoid to set tractionEngineId to null if the onBlur is due to a select change
+    if (!isSelectChanged) {
+      if (filteredRollingStockList.length === 1) {
+        dispatch(updateRollingStockID(filteredRollingStockList[0].id));
+      } else {
+        dispatch(updateRollingStockID(undefined));
+      }
+    }
+    setIsSelectChanged(false);
+  };
+
+  const onSelectSuggestion = (option: StdcmSuggestionsConsistOption) => {
+    dispatch(updateRollingStockID(option.value.id));
+    setIsSelectChanged(true);
+  };
+
+  useEffect(() => {
+    if (rollingStock) {
+      searchMateriel(getLabel(rollingStock));
+    }
+  }, [rollingStock]);
+
+  return (
+    <StdcmCard
+      name="Convoi"
+      title={<ConsistCardTitle rollingStock={rollingStock} />}
+      disabled={isPending}
+    >
```

Comment:
Not sure if its on purpose but on model (screen below), the rolling stock image seems to "ride" on the top of the card body but yours is centered.

![Capture d’écran 2024-05-23 à 15 43 47](https://github.com/OpenRailAssociation/osrd/assets/113104137/20ef0280-f42f-4285-a223-142927815092)